token/vault.go

```diff
@@ -79,3 +79,26 @@ func (v *Vault) NewQueryEngine() *QueryEngine {
 type UnspentTokensIterator struct {
 	driver.UnspentTokensIterator
 }
+
+// Sum consumes the iterator and computes the sum of the quantities of the tokens in the iterator
```

Comment:
maybe we don't use the word consumes?

Reply:
Sum does consume the iterator meaning that after the invocation of Sum. If you call Next, nil is returned